src/api/entities/SecurityToken/__tests__/TokenHolders.ts

```diff
@@ -38,33 +73,102 @@ describe('TokenHolders class', () => {
+
+      const balanceOfEntries: [StorageKey, Balance][] = [];
+
+      const context = dsMockUtils.getContextInstance();
+
+      fakeData.forEach(({ identity, value }) => {
+        const identityId = dsMockUtils.createMockIdentityId(identity);
+        const fakeBalance = dsMockUtils.createMockBalance(value);
+        const balance = new BigNumber(value);
+
+        identityIdToStringStub.withArgs(identityId).returns(identity);
+        balanceToBigNumberStub.withArgs(fakeBalance).returns(balance);
+
+        balanceOfEntries.push(
+          tuple(({ args: [rawTicker, identityId] } as unknown) as StorageKey, fakeBalance)
+        );
+
+        expectedHolders.push({
+          identity: new Identity({ did: identity }, context),
+          balance,
+        });
+      });
+
+      requestPaginatedStub.resolves({ entries: balanceOfEntries, lastKey: null });
+
+      const token = entityMockUtils.getSecurityTokenInstance();
+      const tokenHolders = new TokenHolders(token, context);
+
+      const result = await tokenHolders.get();
+
+      expect(result).toEqual({ data: expectedHolders, next: null });
+    });
 
+    test('should retrieve all the token holders with balance and mint status in false', async () => {
       dsMockUtils.createQueryStub('asset', 'balanceOf');
 
-      const fakeData = [
-        {
-          identity: 'someIdentity',
-          value: 1000,
-        },
-        {
-          identity: 'otherIdentity',
-          value: 2000,
+      const expectedHolders: IdentityBalance[] = [];
+
+      entityMockUtils.configureMocks({
+        securityTokenOptions: {
+          transfersAreFrozen: true,
         },
-      ];
+      });
+
+      const balanceOfEntries: [StorageKey, Balance][] = [];
+
+      const context = dsMockUtils.getContextInstance();
+
+      fakeData.forEach(({ identity, value }) => {
+        const identityId = dsMockUtils.createMockIdentityId(identity);
+        const fakeBalance = dsMockUtils.createMockBalance(value);
+        const balance = new BigNumber(value);
+
+        identityIdToStringStub.withArgs(identityId).returns(identity);
+        balanceToBigNumberStub.withArgs(fakeBalance).returns(balance);
+
+        balanceOfEntries.push(
+          tuple(({ args: [rawTicker, identityId] } as unknown) as StorageKey, fakeBalance)
+        );
+
+        expectedHolders.push({
+          identity: new Identity({ did: identity }, context),
+          balance,
+          canMint: false,
+        });
+      });
+
+      requestPaginatedStub.resolves({ entries: balanceOfEntries, lastKey: null });
+
+      const token = entityMockUtils.getSecurityTokenInstance();
+      const tokenHolders = new TokenHolders(token, context);
+
+      const result = await tokenHolders.get({ mintStatus: true });
+
+      expect(result).toEqual({ data: expectedHolders, next: null });
+    });
+
+    test('should retrieve all the token holders with balance and canMint attribute setting in true', async () => {
```

Comment:
should return whether each token holder can receive primary issuance to if canBeIssuedTo is passed as true